liberty-maven-plugin/src/it/dev-it/src/test/java/net/wasdev/wlp/test/dev/it/DevTest.java

```diff
@@ -208,4 +210,60 @@ public void resolveDependencyTest() throws Exception {
       assertTrue(systemHealthTarget.exists());
    }
 
+   // TODO @Test
+   public void generateFeatureTest() throws Exception {
+
+      final String GENERATED_FEATURES_FILE_NAME = "generated-features.xml";
+      final String SERVER_XML_COMMENT = "Plugin has generated Liberty features"; // the explanation added to server.xml
+      // TODO final String NEW_FILE_INFO_MESSAGE = "Some message"; // the explanation added to the generated features file
+
+      assertTrue(verifyLogMessageExists("Liberty is running in dev mode.", 10000));
+      assertFalse(verifyLogMessageExists("batch-1.0", 10000));
+
+      File newFeatureFile = new File(tempProj, "/src/main/liberty/config/configDropins/overrides/"+GENERATED_FEATURES_FILE_NAME);
+      File newTargetFeatureFile = new File(targetDir, "/liberty/wlp/usr/servers/defaultServer/configDropins/overrides/"+GENERATED_FEATURES_FILE_NAME);
+      File serverXmlFile = new File(tempProj, "/src/main/liberty/config/server.xml");
+      assertTrue(serverXmlFile.exists());
+
+      // Copy a Java file into place to create the HelloBatch class
+      File helloBatchRes = new File("../resources/HelloBatch.java");
+      assertTrue(helloBatchRes.exists());
+      File helloBatchSrc = new File(tempProj, "/src/main/java/com/demo/HelloBatch.java");
+      FileUtils.copyFile(helloBatchRes, helloBatchSrc);
+      assertTrue(helloBatchSrc.exists());
+
+      // Dev mode should now compile the new Java file...
+      File helloBatchObj = new File(tempProj, "target/classes/com/demo/HelloBatch.class");
+      verifyFileExists(newTargetFeatureFile, 15000);
+      // ... and run the proper mojo.
+      assertTrue(verifyLogMessageExists("Running liberty:generate-features", 10000)); // mojo ran
+      assertTrue(verifyFileExists(newFeatureFile, 5000)); // mojo created file
+      assertTrue(verifyFileExists(newTargetFeatureFile, 5000)); // mojo copied file
+      assertTrue(verifyLogMessageExists("batch-1.0", 10000, newFeatureFile));
+      // TODO assertTrue(verifyLogMessageExists(NEW_FILE_INFO_MESSAGE, 10000, newFeatureFile));
+      assertTrue(verifyLogMessageExists(SERVER_XML_COMMENT, 10000, serverXmlFile));
+      assertTrue(verifyLogMessageExists("batch-1.0", 10000)); // should appear in the message "CWWKF0012I: The server installed the following features:"
+   }
+
+   @Test
+   public void mpVersionTest() throws Exception {
```

Comment:
This is testing when we have a specific MicroProfile feature, can we figure out the umbrella version that contains that feature. This is something we have to revisit post snapshot as this may not end up being used this way.